File: internal/model/ar_invoice.go
package model

import (
	"time"

	"github.com/google/uuid"
	"github.com/shopspring/decimal"
)

type ArInvoiceStatus string

const (
	ArInvoiceStatusDraft     ArInvoiceStatus = "draft"
	ArInvoiceStatusConfirmed ArInvoiceStatus = "confirmed"
	ArInvoiceStatusReversed  ArInvoiceStatus = "reversed"
)

type ArInvoice struct {
	ID          uuid.UUID       `db:"id"`
	TenantID    uuid.UUID       `db:"tenant_id"`
	CompanyID   uuid.UUID       `db:"company_id"`
	CustomerID  uuid.UUID       `db:"customer_id"`
	InvoiceID   uuid.UUID       `db:"invoice_id"`
	InvoiceNo   string          `db:"invoice_no"`
	Amount      decimal.Decimal `db:"amount"`
	DueDate     *time.Time      `db:"due_date"`
	Status      string          `db:"status"`
	SourceType  string          `db:"source_type"`
	Remark      *string         `json:"remark,omitempty" db:"remark"`
	CreatedBy   *uuid.UUID      `db:"created_by"`
	CreatedAt   time.Time       `db:"created_at"`
	ConfirmedAt *time.Time      `db:"confirmed_at"`
	ConfirmedBy *uuid.UUID      `db:"confirmed_by"`
	ApprovedBy  *uuid.UUID      `json:"approved_by,omitempty" db:"approved_by"`
	ApprovedAt  *time.Time      `json:"approved_at,omitempty" db:"approved_at"`
}

File: frontend/src/api/modules/ar_invoice.ts
import request from '@/api/request'
import type { ApiResponse } from '@/types/api'

export interface ArInvoice {
  id: string
  invoice_id: string
  invoice_no: string
  customer_id: string
  customer_name?: string
  amount: string
  paid_amount?: string
  outstanding_amount?: string
  due_date?: string | null
  status: string
  source_type: string
  remark?: string
  created_at: string
  confirmed_at?: string | null
  approved_at?: string | null
}

export function fetchArInvoices(params: {
  status?: string
} = {}): Promise<ApiResponse<{ list: ArInvoice[]; total: number }>> {
  return request.get('/ar-invoices', { params })
}

export function fetchArInvoiceById(id: string): Promise<ApiResponse<ArInvoice>> {
  return request.get(`/ar-invoices/${id}`)
}
